modules/v1_14_R1/src/main/java/net/countercraft/movecraft/compat/v1_14_R1/IWorldHandler.java

```diff
@@ -103,16 +105,36 @@ public void rotateCraft(@NotNull Craft craft, @NotNull MovecraftLocation originP
         //*         Step two: Get the tiles         *
         //*******************************************
         WorldServer nativeWorld = ((CraftWorld) craft.getWorld()).getHandle();
-        List<TileHolder> tiles = new ArrayList<>();
+        List<TileEntity> tiles = new ArrayList<>();
+        List<NextTickListEntry> tileTicks = new ArrayList<>();
+        HitBox testBox = craft.getHitBox();
+        BlockPosition position1 = new BlockPosition(testBox.getMidPoint().getX() - testBox.getZLength()/2 - 2, 0, testBox.getMidPoint().getZ() - testBox.getXLength()/2 - 2);
+        BlockPosition position2 = new BlockPosition(testBox.getMidPoint().getX() + testBox.getZLength()/2 + 2, 0, testBox.getMidPoint().getZ() + testBox.getXLength()/2 + 2);
+        List<NextTickListEntry<Block>> capturedTicks = nativeWorld.getBlockTickList().a(new StructureBoundingBox(position1, position2), true, true);
         //get the tiles
         for (BlockPosition position : rotatedPositions.keySet()) {
-            //TileEntity tile = nativeWorld.removeTileEntity(position);
-            TileEntity tile = removeTileEntity(nativeWorld, position);
-            if (tile == null)
+            if (nativeWorld.getType(position) == Blocks.AIR.getBlockData())
                 continue;
-            tile.a(ROTATION[rotation.ordinal()]);
-            //get the nextTick to move with the tile
-            tiles.add(new TileHolder(tile, tickProvider.getNextTick(nativeWorld, position), position));
+            //Find the ticks associated with this location
+            for (NextTickListEntry entry : capturedTicks) {
+                if (entry.a.equals(position)) {
+                    tileTicks.add(entry);
+                }
+            }
+
+            //Remove the old tile and add it to the list
+            TileEntity tile = removeTileEntity(nativeWorld, position);
+            if (tile != null) {
+                tile.a(ROTATION[rotation.ordinal()]);
+                tiles.add(tile);
+            }
+        }
+
+        //Put back all the ticks we didn't move
+        capturedTicks.removeAll(tileTicks);
+        for (NextTickListEntry entry : capturedTicks) {
```

Comment:
`<?>`

Reply:
This is done because the method of removing the tile ticks will only act in a rectangular box from y = 0 to y = 255. This would remove tile ticks below the craft if we did not add them back.